### PR TITLE
CA2018 Analyzer Documentation

### PR DESCRIPTION
**Pull request by mahdiva**
This PR adds the documentation for a Roslyn analyzer that was recently added, CA2018: `Buffer.BlockCopy` expects the number of bytes to be copied for the `count` argument

### CODE REVIEW

File: docs/fundamentals/code-analysis/quality-rules/ca2018.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+| | Value |
+|-|-|
+| **Rule ID** |CA2018|
+| **Category** |[Reliability](reliability-warnings.md)|
+| **Fix is breaking or non-breaking** |Non-breaking|
+
+## Cause
+
+This rule fires when `Array.Length` is used for the `count` argument on arrays with elements larger than 1 byte.
```
Comment (pgovind) [inline docs/fundamentals/code-analysis/quality-rules/ca2018.md:24]: This should at least read as `elements of size larger than 1 byte`. Will leave it up to the docs folks on whether we should mention that `byte[], sbyte[] or bool[]` are the only array types where each element is of size 1 byte.

File: docs/fundamentals/code-analysis/quality-rules/ca2018.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+| **Category** |[Reliability](reliability-warnings.md)|
+| **Fix is breaking or non-breaking** |Non-breaking|
+
+## Cause
+
+This rule fires when `Array.Length` is used for the `count` argument on arrays with elements of size larger than 1 byte.
+
+## Rule description
+
+When using `Buffer.BlockCopy`, the `count` argument expects the number of bytes to be copied. Using `Array.Length` for the `count` argument should only be used for arrays with element size of exactly 1 byte.
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/quality-rules/ca2018.md:28]:
```suggestion
When using `Buffer.BlockCopy`, the `count` argument specifies the number of bytes to copy. You should only use `Array.Length` for the `count` argument on arrays whose elements are exactly one byte in size. `byte`, `sbyte, and `bool` arrays have elements that are one byte in size.
```

File: docs/fundamentals/code-analysis/quality-rules/ca2018.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+## How to fix violations
+
+The `count` argument should be the number of bytes that you intend to copy.
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/quality-rules/ca2018.md:32]:
```suggestion
Specify the number of bytes that you intend to copy for the `count` argument.
```

File: docs/fundamentals/code-analysis/quality-rules/ca2018.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+### Example
+
+```csharp
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/quality-rules/ca2018.md:36]:
```suggestion
**Violation**:

```csharp
```

File: docs/fundamentals/code-analysis/quality-rules/ca2018.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    {
+        int[] src = new int[] {1, 2, 3, 4};
+        int[] dst = new int[] {0, 0, 0, 0};
+
+        Buffer.BlockCopy(src, 0, dst, 0, src.Length); 
+    }
+}
+```
+
+Fix:
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/quality-rules/ca2018.md:50]:
```suggestion
**Fix**:
```

File: docs/fundamentals/code-analysis/quality-rules/ca2018.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+When using arrays with elements of size larger than 1 byte, one possible solution is to multiply the length of the array with the array element size to get the number of bytes.
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/quality-rules/ca2018.md:52]:
```suggestion
If your array's elements are larger than one byte in size, you can multiply the length of the array by the element size to get the number of bytes.
```

**Review by gewarren — APPROVED** [review]

**Comment by gewarren** [comment]
Thank you @mahdiva

**Pull request merged**